### DISCUSSION TITLE
Refetch all cached pages with useInfiniteQuery if only first page's data has changed

### DISCUSSION THREAD

**Discussion by mertafor** [TanStack/query#5693] [Q&A]
I'm using useInfiniteQuery for a chat app on React Native. I pull first 10 messages on load, then fetching other N pages by scrolling down Flatlist. Everything works fine but I also use refetchInterval to refetch messages every 5 seconds. Of course I have several pages (usually have), useInfiniteQuery makes several requests to refetch each page on refetchInterval.

Of course it makes sense but actually refetching all pages only needed if the first page's first item ID is not equal to cached list's first item's ID (or comparing entire object). I can do this by disabling refetchInterval and instead make a separate request, compare item id's and refetch entire list if necessary. But before going into trouble and making big changes in my components, I wanted to ask if there is a better method to achieve this. 

I noticed queryClient.setQueryData option in the documentation however as far as I can see it's best for mutating the cached data.

Edit : I realized increasing query limit and making other adjustments would be much better idea rather than this. By the way huge fan of React Query. Looking for the v5!

**Comment by gupta-ji6** [comment]
you can get some insights from this discussion -  https://github.com/TanStack/query/discussions/1670
